lib/fluent/plugin/in_exec.rb

```diff
@@ -95,6 +95,7 @@ def run(io)
       else
         @parser.parse(io.read, &method(:on_record))
       end
+      io.close if io.eof?
```

Comment:
Generally speaking, resources should be closed by the one who opened it.
I'm not sure but closing it here might not appropriate.